SU2_CFD/include/solvers/CFVMFlowSolverBase.inl

```diff
@@ -0,0 +1,1646 @@
+/*!
+ * \file CFVMFlowSolverBase.hpp
+ * \brief Base class template for all FVM flow solvers.
+ * \version 7.0.6 "Blackbird"
+ *
+ * SU2 Project Website: https://su2code.github.io
+ *
+ * The SU2 Project is maintained by the SU2 Foundation
+ * (http://su2foundation.org)
+ *
+ * Copyright 2012-2020, SU2 Contributors (cf. AUTHORS.md)
+ *
+ * SU2 is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * SU2 is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with SU2. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include "CFVMFlowSolverBase.inl"
+#include "../gradients/computeGradientsGreenGauss.hpp"
+#include "../gradients/computeGradientsLeastSquares.hpp"
+#include "../limiters/computeLimiters.hpp"
+
+
+template<class V, ENUM_REGIME R>
+void CFVMFlowSolverBase<V,R>::AeroCoeffsArray::allocate(int size) {
+  _size = size;
+  CD = new su2double[size]; CL = new su2double[size]; CSF = new su2double[size]; CEff = new su2double[size];
+  CFx = new su2double[size]; CFy = new su2double[size]; CFz = new su2double[size]; CMx = new su2double[size];
+  CMy = new su2double[size]; CMz = new su2double[size]; CoPx = new su2double[size]; CoPy = new su2double[size];
+  CoPz = new su2double[size]; CT = new su2double[size]; CQ = new su2double[size]; CMerit = new su2double[size];
+  setZero();
+}
+
+template<class V, ENUM_REGIME R>
+CFVMFlowSolverBase<V,R>::AeroCoeffsArray::~AeroCoeffsArray() {
+  delete [] CD; delete [] CL; delete [] CSF; delete [] CEff;
+  delete [] CFx; delete [] CFy; delete [] CFz; delete [] CMx;
+  delete [] CMy; delete [] CMz; delete [] CoPx; delete [] CoPy;
+  delete [] CoPz; delete [] CT; delete [] CQ; delete [] CMerit;
+}
+
+template<class V, ENUM_REGIME R>
+void CFVMFlowSolverBase<V,R>::AeroCoeffsArray::setZero(int i) {
+  CD[i] = CL[i] = CSF[i] = CEff[i] = 0.0;
+  CFx[i] = CFy[i] = CFz[i] = CMx[i] = 0.0;
+  CMy[i] = CMz[i] = CoPx[i] = CoPy[i] = 0.0;
+  CoPz[i] = CT[i] = CQ[i] = CMerit[i] = 0.0;
+}
+
+template<class V, ENUM_REGIME R>
+void CFVMFlowSolverBase<V,R>::Allocate(const CConfig& config) {
+
```

Comment:
Implementations of long functions go to the .inl file, we include this file in the cpp where we want the instantiation of the class template to take place (we can spread out the compilation cost like this).